<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>753</width>
    <height>731</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="5,6">
    <item>
     <layout class="QVBoxLayout" name="LeftPanel" stretch="2,8,2,4,1">
      <item>
       <widget class="QGroupBox" name="IpSettings">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>ArrowCursor</cursorShape>
        </property>
        <property name="title">
         <string>Exoskeleton IP</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="cur_ip_label">
           <property name="font">
            <font>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>TextLabel</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_connect">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="Functions">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="title">
         <string>Functions</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2" stretch="4,1,1">
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="0">
            <widget class="QCheckBox" name="checkBox_setExoNeutralPos">
             <property name="text">
              <string>Set Neutral Pos</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QCheckBox" name="checkBox_act_RKne">
             <property name="text">
              <string>Act RKnee</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QCheckBox" name="checkBox_fsm_left_start">
             <property name="text">
              <string>FSM Left Start</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QCheckBox" name="checkBox_rel_RAnk">
             <property name="text">
              <string>Release RAnk</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QCheckBox" name="checkBox_fsm_right_start">
             <property name="text">
              <string>FSM Right Start</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QCheckBox" name="checkBox_rel_RKne">
             <property name="text">
              <string>Release RKnee</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="checkBox_rel_LAnk">
             <property name="text">
              <string>Release LAnk</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_rel_LKne">
             <property name="text">
              <string>Release LKnee</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButton_walkRec">
           <property name="text">
            <string>Walking Energy Recycling</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="btn_sendCmd">
             <property name="text">
              <string>Send</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="but_clearCmd">
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="DataRec">
        <property name="title">
         <string>Data Recording</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="2,1">
         <item>
          <widget class="QLabel" name="label_startTime">
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="textInteractionFlags">
            <set>Qt::TextSelectableByMouse</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPushButton" name="btn_syncTime">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>SYNC TIME</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_rec_start">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>REC Start</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="MainReservoir">
        <property name="title">
         <string>Main Reservoir</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QProgressBar" name="air_volumn">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="value">
            <number>24</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLCDNumber" name="lcd_TankPre"/>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="btn_relax">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Relax</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="btn_lock">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Lock</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="FSM">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>FSM</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcd_fsm"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="RightPanel" stretch="5,2,2">
      <item>
       <widget class="GraphicsLayoutWidget" name="exo_rt_plot"/>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="2" column="2">
         <widget class="QPushButton" name="btn_resetRAnk">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Reset RAnkle</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="btn_resetRKne">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Reset RKnee</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="btn_resetLHip">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string> Reset LHip</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="btn_resetLAnk">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Reset LAnkle</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="btn_resetLKne">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Reset LKnee</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="btn_resetRHip">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Reset RHip</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBox_encoder_reset">
          <property name="text">
           <string>Enable Encoder 
Reset</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="btn_reverse_knee">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Reverse Knee</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Torque (Nm)</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="2">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Right Knee</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Left Knee</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLCDNumber" name="lcd_right_knee_tor"/>
         </item>
         <item row="0" column="1">
          <widget class="QLCDNumber" name="lcd_left_knee_tor"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Left Ankle</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLCDNumber" name="lcd_left_ank_tor"/>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Right Ankle</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLCDNumber" name="lcd_right_ank_tor"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>753</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConnection">
    <property name="title">
     <string>Connection</string>
    </property>
    <addaction name="actionSet_IP_Address"/>
   </widget>
   <widget class="QMenu" name="menuGraphs">
    <property name="title">
     <string>Monitor</string>
    </property>
    <addaction name="actionGraphJoint"/>
    <addaction name="action_plotPressure"/>
    <addaction name="actionPWM_Duty"/>
    <addaction name="actionPhase"/>
   </widget>
   <widget class="QMenu" name="menuTask_Settings">
    <property name="title">
     <string>Robot Settings</string>
    </property>
    <addaction name="actionCylinder_Length"/>
    <addaction name="actionFSM_Parameter"/>
   </widget>
   <widget class="QMenu" name="menuAction">
    <property name="title">
     <string>Test Functions</string>
    </property>
    <addaction name="actionTest_SW_Valves"/>
    <addaction name="act_testPWM_valves"/>
    <addaction name="actionTest_Pressure_Control"/>
    <addaction name="act_ImpCon"/>
    <addaction name="action_ImpactAbsorb"/>
   </widget>
   <addaction name="menuConnection"/>
   <addaction name="menuGraphs"/>
   <addaction name="menuTask_Settings"/>
   <addaction name="menuAction"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSet_IP_Address">
   <property name="text">
    <string>Set IP Address</string>
   </property>
  </action>
  <action name="actionGraphJoint">
   <property name="text">
    <string>Joint Angles</string>
   </property>
  </action>
  <action name="action_plotPressure">
   <property name="text">
    <string>Pressure</string>
   </property>
  </action>
  <action name="actionTest_SW_Valves">
   <property name="text">
    <string>Test SW Valves</string>
   </property>
  </action>
  <action name="act_testPWM_valves">
   <property name="text">
    <string>Test PWM Valves</string>
   </property>
  </action>
  <action name="actionPWM_Duty">
   <property name="text">
    <string>PWM Duty</string>
   </property>
  </action>
  <action name="actionTest_MPC">
   <property name="text">
    <string>Test MPC</string>
   </property>
  </action>
  <action name="act_ImpCon">
   <property name="text">
    <string>Test Impedance Control</string>
   </property>
  </action>
  <action name="actionTest_Pressure_Control">
   <property name="text">
    <string>Test Pressure Control</string>
   </property>
  </action>
  <action name="action_ImpactAbsorb">
   <property name="text">
    <string>Test Impact Absorb</string>
   </property>
  </action>
  <action name="actionCylinder_Length">
   <property name="text">
    <string>Cylinder Length</string>
   </property>
  </action>
  <action name="actionFSM_Parameter">
   <property name="text">
    <string>FSM Parameter</string>
   </property>
  </action>
  <action name="actionForce_Torque">
   <property name="text">
    <string>Force/Torque</string>
   </property>
  </action>
  <action name="actionPhase">
   <property name="text">
    <string>Phase </string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GraphicsLayoutWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
